### PR TITLE
python: adapt for new Python module versions

### PR DESCRIPTION
**Pull request by joergsteffens**
This PR makes some changes, so that python-bareos and bareos-restapi are able to be used with the current versions of the Python modules `packaging` (tested with 23.1) and `pydantic` (tested with 2.3.0).

The pydantic problem has been reported by #1544 

**Backport of PR #0000 to bareos-2x** (remove this line if this is no backport; for backport use cherry-pick -x)

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)

##### Tests
- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: systemtests/tests/restapi/api/restapi.sh [hunk 1]
```diff
@@ -34,12 +34,14 @@ else
     export PYTHONPATH=${CMAKE_SOURCE_DIR}/restapi/
 fi
 
+LSOF_CMD="lsof -n -iTCP:$REST_API_PORT"
```
Comment (sduehr) [inline systemtests/tests/restapi/api/restapi.sh:37]:
```suggestion
LSOF_CMD="lsof -n -iTCP:$REST_API_PORT -sTCP:LISTEN"
```

Shouldn't we add `-sTCP:LISTEN` here?
I think that only the listening state is relevant here, and looks like `lsof` doesn't distinguish between source and target port.

**Review by sduehr — CHANGES_REQUESTED** [review]
Thanks, looks good. Only one small suggestion.

**Review by sduehr — APPROVED** [review]
Thanks, looks good now.

**Comment by joergsteffens** [comment]
@sduehr I integrated your suggestion. Please check again.

**Review by sduehr — APPROVED** [review]
Thanks

**Pull request merged**